From a4f7662a4347450b0af8ece3ae90ff0dcb70bf6e Mon Sep 17 00:00:00 2001
From: Johan Engelen <jbc.engelen@gmail.com>
Date: Sat, 25 Nov 2023 21:57:33 +0100
Subject: [PATCH 10/17] Fixup compiler-rt dylib codesign signature. Needed to
 make ASan work well (without it, no report is printed).

---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5c86891277..304ef21878 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -803,6 +803,7 @@ function(copy_compilerrt_lib llvm_lib_name ldc_lib_name fixup_dylib)
         copy_and_rename_file(${llvm_lib_path} ${ldc_lib_path})
         if (fixup_dylib)
             execute_process(COMMAND install_name_tool -id @rpath/${ldc_lib_name} ${ldc_lib_path})
+            execute_process(COMMAND codesign --force -s - ${ldc_lib_path})
         endif()
         install(FILES ${ldc_lib_path} DESTINATION ${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX})
     else()
-- 
2.39.1

